tests/lock.cpp

```diff
@@ -36,91 +38,92 @@ constexpr int iter = 10000;
+		int *counter;
+
+		LockTest() {
+			argo::reset();
+			field = argo::conew_<tas_lock::internal_field_type>();
+			global_tas_lock = new tas_lock(field);
+			argo::barrier();
+		}
 
-protected:
-	/** @brief local ticket lock for testing */
-	argo::locallock::ticket_lock *ticket_lock;
-	/** @brief global MCS lock for testing */
-	argo::locallock::mcs_lock *mcs_lock;
-	/** @brief global cohort lock for testing */
-	argo::globallock::cohort_lock *cohort_lock;
-	/** @brief global tas lock type */
-	using tas_lock = argo::globallock::global_tas_lock;
-	/** @brief global tas lock for testing*/
-	tas_lock *global_tas_lock;
-	/** @brief field needed for the global tas lock*/
-	tas_lock::internal_field_type* field;
-	/** @brief global counter used in several tests */
-	int *counter;
-
-	LockTest() {
-		argo::reset();
-		field = argo::conew_<tas_lock::internal_field_type>();
-		global_tas_lock = new tas_lock(field);
-		argo::barrier();
-	}
-
-	~LockTest() {
-		argo::codelete_(field);
-		delete global_tas_lock;
-		argo::barrier();
-	}
+		~LockTest() {
+			argo::codelete_(field);
+			delete global_tas_lock;
+			argo::barrier();
+		}
 };
 
+
 /**
  * @brief Tests trylock functionality
  */
-TEST_F(LockTest,TAS_trylock_all) {
+TEST_F(LockTest, TAS_trylock_all) {
 	bool *did_increment;
 	bool res;
 	counter = argo::conew_<int>();
 	did_increment = argo::conew_array<bool>(argo::number_of_nodes());
 
 	/* Init a counter and increment array to 0 */
-	if(argo::node_id() == 0){
+	if(argo::node_id() == 0) {
 		*counter = 0;
-		for (argo::num_nodes_t i = 0; i < argo::number_of_nodes(); i++){
+		for (argo::num_nodes_t i = 0; i < argo::number_of_nodes(); i++) {
 			did_increment[i] = false;
 		}
 	}
 	argo::barrier();
 
 	/* All nodes try to take the lock and increment a shared counter */
 	ASSERT_NO_THROW(res = global_tas_lock->try_lock());
-	if(res){
+	if(res) {
 		(*counter)++;
 		did_increment[argo::node_id()] = true;
 		ASSERT_NO_THROW(global_tas_lock->unlock());
 	}
 	argo::barrier();
 
 	/* At least not more than number_of_nodes nodes can have incremented the counter */
-	ASSERT_GE(argo::number_of_nodes(),*counter);
+	ASSERT_GE(argo::number_of_nodes(), *counter);
 
 	int sum = 0;
-	for (argo::num_nodes_t i = 0; i < argo::number_of_nodes(); i++){
+	for (argo::num_nodes_t i = 0; i < argo::number_of_nodes(); i++) {
 		if(did_increment[i])
 			sum++;
 	}
 
 	/* Compares counter value to flags set by each node */
-	ASSERT_EQ(sum,*counter);
+	ASSERT_EQ(sum, *counter);
 
 	argo::codelete_(counter);
 	argo::codelete_array(did_increment);
 }
 
-
-/** @brief Checks locking is working by implementing a custom barrier */
-TEST_F(LockTest,TAS_lock_custom_barrier) {
+/**
+ * @brief Checks locking is working by implementing a custom barrier
+ */
+TEST_F(LockTest, TAS_lock_custom_barrier) {
 	unsigned int tmp;
 
 	/* Checks this many times if barrier was successful */
-	long deadlock_threshold = 100000;
+	std::size_t deadlock_threshold = 100000;
```

Comment:
Cpplint gave me the following for this one:
```sh
tests/lock.cpp:122:  Use int16/int64/etc, rather than the C type long  [runtime/int] [4]
```
and since we do:
https://github.com/etascale/argodsm/blob/bdc6514d0803305ba28c933d717dc4ce48cb0bf6/tests/lock.cpp#L139-L142
I thought it should be fine to use `std::size_t` here.

But changing this to an `int` instead of `long`, cpplint doesn't spit anything, so I guess we can go with that.